examples/solvers/functional_basic_example.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Basic example on how to write a functional.
+
+When defining a new functional, there are a few standard methods and properties
+that can be implemeted. These are:
+- ``__init__``. This method intialize the functional
+- ``_call``. The actual function call ``functional(x)``
+- ``gradient`` (property). This gives the gradient operator of the functional,
+    i.e., the operator that corresponds to the mapping ``x -> grad_f(x)``
+- ``proximal``. This returns the proximal operator. If called only as
+    ``functional.proximal``, it corresponds to a `Proximal factory`.
+- ``conjugate_functional`` (property). This gives the convex conjugate
+    functional
+- ``derivative``. This returns the (directional) derivative operator in a point
+    y, such that when called with a point x it corresponds to the linear
+    operator ``x --> <x, grad_f(y)>``. Note that this has a default
+    implemetation that uses the gradient in order to achieve this.
+
+Below follows an example of implementing the functional ``||x||_2^2 + <x,y>``,
+for some parameter ``y.``
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+import odl
+
+# Here we define the functional
+class MyFunctional(odl.solvers.Functional):
+    """This is my functional: ||x||_2^2 + <x, y>."""
+
+    # Defining the __init__ function
+    def __init__(self, domain, y):
```

Comment:
Even for an example, a meaningless name like `y` is not good. It's harder to remember and easier to get wrong than something with a meaning.